model/riscv_insts_fext.sail

```diff
@@ -1056,16 +1056,16 @@ function clause execute (F_UN_TYPE_S(rs1, rd, FCLASS_S)) = {
 }
 
 function clause execute (F_UN_TYPE_S(rs1, rd, FMV_X_W)) = {
-  let rs1_val_S            = F(rs1)[31..0];
-  let rd_val_X  : xlenbits = sign_extend(rs1_val_S);
-  X(rd) = rd_val_X;
+  // A narrower n-bit transfer out of the floating-point registers will
+  // transfer the lower n bits of the register ignoring the upper FLEN-n bits.
```

Comment:
I don't see the point of these comments? These behave exactly as one would expect, and are entirely consistent with the rest of the ISA.

Reply:
That is, reading a 32-bit value reads that 32-bit sub register, and writing one either sign-extends, in the case of integer destinations, or nan boxes, in the case of float destinations.

Reply:
Yeah it wasn't meant to contradict any expectations - it's a direct quote from the spec; just explain why the code was doing what it is (because it's not obvious IMO unless you are an expert). I can remove it if you like. Or maybe put it in quotes?

Reply:
I would delete it. The existence of the comment suggest something different to what is expected / done elsewhere, but it is entirely consistent and so to me increases confusion. This code is entirely what someone would expect for an instruction that is “move a 32-bit float to an X register”.

Reply:
(and similarly for the other direction below)

Reply:
Ok, deleted.